src/LanguageServer/Impl/Implementation/Server.Completion.cs

```diff
@@ -70,39 +69,50 @@ public sealed partial class Server {
                 _allowSnippet = ctxt.ShouldAllowSnippets
             };
 
+            res._applicableSpan = GetApplicableSpan(ctxt, @params, tree);
+            LogMessage(MessageType.Info, $"Found {res.items.Length} completions for {uri} at {@params.position} after filtering");
+
+            if (HandleOldStyleCompletionExtension(analysis as ModuleAnalysis, tree, @params.position, res)) {
+                return res;
+            }
+
+            await InvokeExtensionsAsync((ext, token)
+                => (ext as ICompletionExtension)?.HandleCompletionAsync(uri, analysis, tree, @params.position, res, cancellationToken), cancellationToken);
+
+            return res;
+        }
+
+        private SourceSpan? GetApplicableSpan(CompletionAnalysis ca, CompletionParams @params, PythonAst tree) {
```

Comment:
Does any of it affects `textEdit` field in the individual `CompletionItem`?

Reply:
Technically they should all have the same applicable span. The span is actually used only by VS to figure out what to delete and replace with the completion item insertion text.